docs/notebooks/How_JAX_primitives_work.ipynb

```diff
@@ -0,0 +1,1651 @@
+            "|<- multiply_add_transpose = (None, Traced<ShapedArray(float32[])>, 1.0)\n",
+            "call multiply_add_xla_translation(<JaxComputationBuilder>, <XlaOp at 0x7f44cfec1b20>, <XlaOp at 0x7f44cfec13e8>, <XlaOp at 0x7f44cfec1880>)\n",
+            "|<- multiply_add_xla_translation = <XlaOp at 0x7f44cfec1420>\n",
+            "call multiply_add_xla_translation(<JaxComputationBuilder>, <XlaOp at 0x7f44cfec13e8>, <XlaOp at 0x7f44cfec1308>, <XlaOp at 0x7f44cfec1ae8>)\n",
+            "|<- multiply_add_xla_translation = <XlaOp at 0x7f44cfec1490>\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "-3lqPkdQPvl5",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Batching\n",
+        "\n",
+        "The batching transformation takes a point-wise computation and turns it\n",
+        "into a computation on vectors. If we try it right now, we get a `NotImplementedError`:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hFvBR3I9Pzh3",
+        "colab_type": "code",
+        "outputId": "434608bc-281f-4d3b-83bd-eaaf3b51b1cd",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 360
+        }
+      },
+      "source": [
+        "# The arguments are two vectors instead of two scalars\n",
+        "with expectNotImplementedError():\n",
+        "  api.vmap(square_add_prim, in_axes=0, out_axes=0)(onp.array([2., 3.]),\n",
+        "                                               onp.array([10., 20.]))"
+      ],
+      "execution_count": 22,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "call square_add_prim(Traced<ShapedArray(float32[])>, Traced<ShapedArray(float32[])>)\n",
+            "  call multiply_add_prim(Traced<ShapedArray(float32[])>, Traced<ShapedArray(float32[])>, Traced<ShapedArray(float32[])>)\n",
+            "\n",
+            "Found expected exception:\n"
+          ],
+          "name": "stdout"
+        },
+        {
+          "output_type": "stream",
+          "text": [
+            "Traceback (most recent call last):\n",
+            "  File \"/usr/local/lib/python3.6/dist-packages/jax/interpreters/batching.py\", line 163, in get_primitive_batcher\n",
+            "    return primitive_batchers[p]\n",
+            "KeyError: multiply_add\n",
+            "\n",
+            "During handling of the above exception, another exception occurred:\n",
+            "\n",
+            "Traceback (most recent call last):\n",
+            "  File \"<ipython-input-22-70154d0e2ab6>\", line 3, in <module>\n",
+            "    onp.array([10., 20.]))\n",
+            "  File \"/usr/local/lib/python3.6/dist-packages/jax/api.py\", line 611, in batched_fun\n",
+            "    lambda: _flatten_axes(out_tree(), out_axes))\n",
+ 
```

Comment:
```suggestion
        "We need to tell JAX how to evaluate the batched version of the primitive. In this particular case, the `multiply_add_prim` already operates pointwise for any dimension of input vectors. So the batched version can use the same `multiply_add_prim` implementation."
```

Reply:
Oops, I think the "suggested change" interface broke here. This was another typofix, multiple_add_prim -> multiply_add_prim.